utilities/zookeeper/R/correctors.county.R

```diff
@@ -0,0 +1,171 @@
+#' @param time_value_flag_date no corrections after this date
+#' @param multinomial_preprocessor logical, do we run the preporcessor
+#' @param corrections_db_path path to database if storing results
+#' @param integer_tol small number to handle integer checks
+#' @param ... ignored
+#'
+#' @return A list of paramter values
+#' @export
+#'
+#' @examples
+#' default_county_params(window_size=21)
+default_county_params <- function(signals_to_correct = "response",
+                                 window_size = 14,
+                                 backfill_lag = 30,
+                                 excess_cut = 0,
+                                 size_cut = 20,
+                                 sig_cut = 3,
+                                 sig_consec = 2.25,
+                                 time_value_flag_date = Sys.Date() + 1,
+                                 multinomial_preprocessor = TRUE,
+                                 corrections_db_path = NULL,
+                                 integer_tol = 1e-6,
+                                 ...) {
+  list(
+    signals_to_correct = signals_to_correct,
+    window_size = window_size,
+    backfill_lag = backfill_lag,
+    excess_cut = excess_cut,
+    size_cut = size_cut,
+    sig_cut = sig_cut,
+    sig_consec = sig_consec,
+    time_value_flag_date = time_value_flag_date,
+    multinomial_preprocessor = multinomial_preprocessor,
+    integer_tol = integer_tol
+  )
+}
+
+#' Corrections function for aardvark
+#'
+#' @param signals_list list of signals as returned from `covidcast_signals()`
+#'   uses the list (no aggregation)
+#' @param ... named corrections parameters passed to `default_county_params()`
+#'
+#' @return a list of signals the same length and format as the input with
+#'   updated entries in the `value` column
+#'
+#'   If the parameter `corrections_db_path` gives a path, intermediate output
+#'   is written to that sqlite database
+#' @export
+zyzzyva_county_corrections <- function(signals_list, ...){
+  if (class(signals_list)[1] == "covidcast_signal") {
+    signals_list <- list(signals_list)
+  }
+  
+  params <- default_county_params(...)
+  len_params <- sapply(params, length)
+  max_len_params <- max(len_params)
+  in_names <- names(signals_list[[1]])
+  assert_that(all(len_params %in% c(1L, max_len_params)),
+              msg = paste("In apply_corrections: ",
+                          "corrections parameters must be length 1 or the",
+                          "same length as the longest corrections parameter."))
+  #Single Signal
+  if (params$signals_to_correct == "response") {
+    corrected <- zyzzyva_county_corrections_single_signal(
+      signals_list[[1]], params)
+    signals_list[[1]] <- corrected %>%
+      mutate(value = .data$corrected) %>%
+      select(all_of(in_names))
+  }
+  
+  # Multiple Signals
+  if (params$signals_to_correct == "all") {
+    corrected <- list()
+    for (i in seq_along(signals_list)) {
```

Comment:
I don't think this will work. We don't want "all" here, just cases and deaths at the moment. So it needs to make sure it finds those.